docs/architecture/adr-001-redesign-knuu-async-approach.md

```diff
@@ -0,0 +1,49 @@
+# ADR 001: Redesigning Knuu for Asynchronous Testing Approach
+
+## Status
+
+Draft
+
+## Context
+
+The existing design of `Knuu` requires exposing the addresses and ports of the target instance running alongside BitTwister, complicating the configuration process. This setup requires Knuu to orchestrate the instances, configure them, and execute the test, leading to complexities in managing traffic shaping. Moreover, as the current model relies on port forwarding, scalability issues have surfaced, causing inconsistent errors and hindering seamless test execution.
+
+## Problem
+
+The current synchronous model necessitates exposing the target instance's address and ports to BitTwister via a sidecar. This centralized approach introduces dependencies and complexities in configuring the test environment. Additionally, the reliance on a proxy approach or port forwarding poses scalability issues, resulting in inconsistent errors during test execution.
+
+## Proposed Solution
+
+### A Distributed Approach with a Playbook Model
+
+Embrace a distributed playbook model similar to Testground, allowing asynchronous orchestration and execution of test scenarios:
+
+1. **Test Scenario Definition:** Users define test scenarios in a descriptive language (e.g., Golang).
```

Comment:
CI/CD gateway is needed here as most of tests should be cron jobs for verification pipelines for upcoming releases of artifacts owned by the contributors, etc.

Reply:
i'd propose looking at something cue for conifg https://cuelang.org/, a bit like HCL but with a bit more dynamism, dagger uses it for it's pipelines and it looks quite nice